### ISSUE TITLE
collection.count 

### ISSUE THREAD

**Issue by rducom** [LuccaSA/RestDrivenDomain#186]
Actuellement nous utilisons l'approche `collection.count` pour obtenir le nombre d'éléments sur une ressource.
L'implémentation actuelle est un cas exceptionnel dans le coded géré pour cette feature.

Je propose de créer une route dédiée sur les controllers de base `/api/Resource/Count` pour atteindre le même périmètre fonctionnel.
De la même façon, nous pourrons étendre à `/api/Resource/Any`.

**Comment by Poltuu** [comment]
je suis d'accord que l'implémentation actuelle est associée à de très nombreuses exceptions peu élégantes tout au long du code concerné.

J'aime beaucoup l'idée d'avoir une route par défaut en plus /count, et également  un /any, mais attention au théorique conflit avec les routes potentiellement existantes.

In any case, je pense qu'on devrait le proposer dans une v3.X plus que dans la v3, dont j'aimerai clore le périmètre fonctionnel soon.

**Comment by nfaugout** [comment]
De quel cas exceptionnel tu parles @rducom ? Le fait d'avoir différencier les fields sur les entités et sur les collections me semble satisfaisant non ?

Créer une sous route adhoc pour ça ne me convient pas, même si y'a pas de conflit avec d'autres routes.

Il faut qu'on garde la logique des URL comme suit :

- /api/{resourceName}/{id}/{command or subResources}

Mais pas des mots adhoc pour obtenir des informations adhoc.

Au pire si vraiment `collection.count` ne vous convient pas, je veux bien qu'on implémente le paging HTTP via le Header links : https://tools.ietf.org/html/rfc5988#page-6

Le seul pb avec les headers, c'est qu'on ne peut pas "jouer" avec l'API via son navigateur..

**Comment by rducom** [comment]
> De quel cas exceptionnel tu parles
Le query string `collection.count` est actuellement très peu utilisé, mais il est utilisé.
Je souhaite trouver une alternative à ce truc, ou à défaut nous en débarrasser purement et simplement.

On pourrait par exemple dire que le Count total étant systématiquement présent lors d'un appel de collection via paging, on peu se débarrasser de ce `collection.count`.

Est-ce que c'est suffisant ?

**Comment by nfaugout** [comment]
Tu veux dire calculer le count systématiquement même s'il n'est pas demandé dans les fields ?

On pourrait revenir à ce qu'on faisait dans la v2 sur Lucca, càd lier le count au paging, car dans les usages les 2 sont intimements liés (tu veux avoir le count total parce que tu veux paginer les résultats). Et si tu voulais "juste" le count total, tu peux faire `&paging=0,1` ça te renverra 1 seul élément ET le count total.

![chrome_2018-10-09_09-41-59](https://user-images.githubusercontent.com/2686141/46653895-9552ab00-cba7-11e8-910b-394707c25de8.png)

**Comment by rducom** [comment]
Exactement ! Voir même `&paging=0,0`
Si cette option te conviens, alors elle me conviens aussi
